metriq_gym/benchmarks/clops.py

```diff
@@ -0,0 +1,155 @@
+        available_edges.remove(edge)
+        edges_to_delete = []
+        for ce in list(available_edges):
+            if (edge[0] in ce) or (edge[1] in ce):
+                edges_to_delete.append(ce)
+        available_edges.difference_update(set(edges_to_delete))
+        if "ecr" in basis_gates:
+            qc.ecr(*edge)
+        elif "cz" in basis_gates:
+            qc.cz(*edge)
+        else:
+            qc.cx(*edge)
+    return
+
+
+# adapted from submodules/qiskit-device-benchmarking/qiskit_device_benchmarking/clops/clops_benchmark.py::prepare_clops_circuits
+def prepare_clops_circuits(
+    width: int,
+    layers: int,
+    num_circuits: int,
+    basis_gates,
+    topology_graph: rx.PyGraph,
+    total_qubits: Optional[int],
+    seed: int = 0,
+) -> list[QuantumCircuit]:
+    qubit_map = create_qubit_map(width, topology_graph, faulty_qubits=[], total_qubits=total_qubits)
+
+    for edge in topology_graph.edge_list():
+        if edge[0] not in qubit_map or edge[1] not in qubit_map:
+            topology_graph.remove_edge(*edge)
+
+    qc = QuantumCircuit(max(qubit_map) + 1, max(qubit_map) + 1)
+    qubits = [qc.qubits[i] for i in qubit_map]
+
+    parameters = []
+    rng = np.random.default_rng(seed)
+    for d in range(layers):
+        append_2q_layer(qc, topology_graph, basis_gates, rng)
+        parameters += append_1q_layer(qc, qubits, parameterized=True, parameter_prefix="L" + str(d))
+
+    for idx in range(width):
+        qc.measure(qubit_map[idx], idx)
+
+    # Parameters are instantiated for each circuit with random values
+    parametrized_circuits = [
+        qc.assign_parameters(
+            [rng.uniform(0, np.pi * 2) for _ in range(sum([len(param) for param in parameters]))]
+        )
+        for _ in range(num_circuits)
+    ]
+
+    return parametrized_circuits
+
+
+class Clops(Benchmark):
+    def dispatch_handler(self, device: QuantumDevice) -> ClopsData:
+        topology_graph = device_topology(device)
+        circuits = prepare_clops_circuits(
+            width=self.params.width,
+            layers=self.params.num_layers,
+            num_circuits=self.params.num_circuits,
+            basis_gates=device.profile.basis_gates or [],
+            topology_graph=topology_graph,
+            total_qubits=device.num_qubits,
+        )
+        quantum_job: QuantumJob | list[QuantumJob] = device.run(circuits, shots=self.params.shots)
+        provider_job_ids = flatten_job_ids(quantum_job)
+        return ClopsData(
+            provider_job_ids=provider_job_ids,
+        )
+
+    def poll_handler(
+        self,
+        job_data: BenchmarkData,
+        result_data: list[GateModelResultData],
+        quantum_jobs: list[QuantumJob],
+    ) -> ClopsResult:
+        if not isinstance(job_data, ClopsData):
+            raise TypeError(f"Expected job_data to be of type {type(ClopsData)}")
+        clops_score = (self.params.num_layers * self.params.shots * self.params.num_circuits) / sum(
```

Comment:
Is there a reference to point to for where this calculation comes from? Maybe it's just obvious from the definition and by virtue of it being in CLOPS it's already self-evident? Thinking out loud, feel free to ignore that one.